File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String alphaStr = s.trim().toLowerCase().replaceAll("[^a-z0-9]","");
        char[] alphaCharArr = alphaStr.toCharArray();
        int start,end;
        start = 0;
        end = alphaCharArr.length -1;
        while(start < end){
            if(alphaCharArr[start] != alphaCharArr[end]){
                return false;
            }
            start++;
            end--;
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        s = s.lower()
        s = ''.join(char for char in s if char.isalnum())
        i,j = 0,len(s)-1
        print(s)
        while i<j:
            if s[i]!=s[j]:
                return False
            i+=1
            j-=1

        return True